apps/meteor/client/providers/CallProvider/CallProvider.tsx

```diff
@@ -149,8 +149,8 @@ export const CallProvider: FC = ({ children }) => {
 
 		setQueueAggregator(voipClient.getAggregator());
 
-		return (): void => voipClient.unregister();
-	}, [result]);
+		return (): void => (clientState === 'registered' ? voipClient.unregister() : undefined);
```

Comment:
I would probably use `clientState === 'registered' && voipClient.unregister()` as a one liner or even better do `if (clientState === 'registered') voipClient.unregister()`. I think it expresses better that the important is the fn call and not the returned value.